bundles/net.tourbook/src/net/tourbook/ui/views/tourDataEditor/TourDataEditorView.java

```diff
@@ -2468,6 +2471,27 @@ public void propertyChange(final PropertyChangeEvent event) {
                // updateUITab4Info(); do NOT work
                //
                // tour data must be reloaded
+            } else if (property.equals(ITourbookPreferences.TOGGLE_STATE_SELECT_INBETWEEN_TIME_SLICES)) {
+
+               _isSelectInBetweenTimeSlices = _prefStore.getBoolean(ITourbookPreferences.TOGGLE_STATE_SELECT_INBETWEEN_TIME_SLICES);
+
+               final Table table = (Table) _timeSlice_Viewer.getControl();
+               final int[] selectionIndices = table.getSelectionIndices();
+
+               if (selectionIndices.length == 0) {
+                  return;
+               }
+
+               table.deselectAll();
+
+               if (_isSelectInBetweenTimeSlices) {
+                  final int minSelectedValue = Math.min(_lastSelectedChartInfo.leftSliderValuesIndex, _lastSelectedChartInfo.rightSliderValuesIndex);
+                  final int maxSelectedValue = Math.max(_lastSelectedChartInfo.leftSliderValuesIndex, _lastSelectedChartInfo.rightSliderValuesIndex);
+                  table.select(minSelectedValue, maxSelectedValue);
+               } else {
+                  table.select(selectionIndices[0]);
+               }
+               table.showSelection();
             }
```

Comment:
Why have you implemented this feature with a pref store field ?

If I would have implemented all property changes with pref store fields there would be many of them. In this case, I would have create a new field in `SelectionChartInfo` with the state `isSelectInBetweenTimeSlices` and may be also in `SelectionChartXSliderPosition`

Reply:
Because I saw that [VALUE_POINT_TOOL_TIP_IS_VISIBLE ](https://github.com/wolfgang-ch/mytourbook/blob/d2cdac9cc46d89e6948e4fe2c90ab6a19f253998/bundles/net.tourbook/src/net/tourbook/preferences/ITourbookPreferences.java#L541) was implemented in the pref store too

I will change based on your recommendation

Reply:
Sometimes it is OK when using the pref store field but in this case it can be avoided.

Reply:
After thinking about it, and to offer to the user a better experience, I have changed the UI for this feature.